stream.js

```diff
@@ -12,18 +12,13 @@ function buildProxy (options, socketWrite, socketEnd) {
     objectMode: options.objectMode
   })
 
-  proxy._destroyed = false
   proxy._write = socketWrite
   proxy._flush = socketEnd
 
-  proxy.destroy = function(err) {
-    if (this._destroyed) return
-      this._destroyed = true
-
+  proxy._destroy = function(err, cb) {
```

Comment:
Using the default implementation with

```
self.push(null)
self.end()
```

is probably better but I didn't change it as there might be something I'm missing here.